modules/osint/domain_reputation.py

```diff
@@ -0,0 +1,153 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+import concurrent.futures
+
+meta = {
+	'name': 'Domain Reputation',
+	'author': 'Kunal Khandelwal',
+	'version': '0.1',
+	'description': 'Check domain reputation with different sources and provide a summary of combined results.',
+	'sources': ('barracudacentral', 'mxtoolbox', 'multirbl'),
+	'options': (
+		('domain', None, True, 'Domain name without https?://', '-d', 'store', str),
+		('engines', 'barracudacentral,multirbl,mxtoolbox', True, 'Search engine names(default=[barracudacentral,multirbl])'
+		 , '-e', 'store', str),
+		('thread', 2, False, 'The number of engine that run per round(default=2)', '-t', 'store', int),
+	),
+	'examples': ('domain_reputation -d example.com',
+				 'domain_reputation -d 93.184.216.34'
+				 )
+}
+
+BLACKLIST = []
+OUTPUT = {}
+LISTS = 0
+
+def thread(self, function, thread_count, engines, q, sources):
+	threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=thread_count)
+	futures = (threadpool.submit(
+		function, self, name, q) for name in engines if name in sources)
+	for _ in concurrent.futures.as_completed(futures):
+		pass
+
+def search(self, name, q):
+	engine = eval(name)
+	engine(self, q)
+
+def barracudacentral(self, q):
+	global LISTS,OUTPUT
+	self.verbose('[Barracuda] Searching in barracuda...')
+	data = {
+		'lookup_entry': q,
+		'submit': 'Check Reputation'
+	}
+	try:
+		req = self.request('https://www.barracudacentral.org/lookups/lookup-reputation', method='POST',
+						   data=data)
+	except Exception as e:
+		self.error('Barracuda is missed!')
+	else:
+		reg = re.compile(r"categories: <strong>([\w\d\s-]+)")
+		if 'listed as "poor"' not in req.text:
+			category = reg.findall(req.text)[0]
+			OUTPUT['category'] = category
+		else:
+			BLACKLIST.append('BARRACUDA')
+		LISTS += 1
+
+def mxtoolbox(self, q):
+	global LISTS
+	self.verbose('[MXTOOLBOX] Searching in mxtoolbox...')
+	try:
+		headers = {'TempAuthorization': 'ca1e3b2f-cd6c-4886-9446-88f4ff7a960d',
+				   'Referer': f'https://mxtoolbox.com/SuperTool.aspx?action=blacklist:{q}&run=toolpage',
+				   'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 '
+								 '(KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36'
+				   }
+		req = self.requests(f'http://mxtoolbox.com/api/v1/Lookup?command=blacklist&argument={q}&resultIndex=1'
```

Comment:
Here, It should be self.request . But I used the requests module because the implementation of def request in [core/core.py](https://github.com/saeeddhqan/Maryam/blob/fa11ee231772a8755240c42bd00ec9d5f4ce04cb/core/core.py#L483) uses the title method over headers which changes the casing of headers. And in this case the server was sensitive to case. Therfore it required **TempAuthorization** in headers but if we use self.request then it would become **Tempauthorization** resulting in breaking the request

Reply:
Got it.